observations/forward_operators/obs_def_abs_humidity_mod.f90

```diff
@@ -0,0 +1,186 @@
+!            continue
+! END DART PREPROCESS INTERACTIVE_OBS_DEF
+
+! BEGIN DART PREPROCESS MODULE CODE
+module obs_def_abs_humidity_mod
+
+use        types_mod, only : r8, missing_r8, gas_constant, gas_constant_v
+use    utilities_mod, only : register_module, error_handler, E_ERR, E_MSG, E_ALLMSG
+use     location_mod, only : location_type, set_location, get_location, write_location, &
+                             read_location, is_vertical
+use  assim_model_mod, only : interpolate
+use     obs_kind_mod, only : QTY_TEMPERATURE, QTY_PRESSURE, QTY_VAPOR_MIXING_RATIO
+
+use ensemble_manager_mod,  only : ensemble_type
+use obs_def_utilities_mod, only : track_status
+
+
+implicit none
+private
+
+public :: get_expected_absolute_humidity
+
+! version controlled file description for error handling, do not edit
+character(len=256), parameter :: source   = &
+   "$URL$"
+character(len=32 ), parameter :: revision = "$Revision$"
+character(len=128), parameter :: revdate  = "$Date$"
+
+logical, save       :: module_initialized   = .false.
+logical, save       :: first_time_warn_low  = .true.
+logical, save       :: first_time_warn_high = .true.
+character(len=512)  :: msgstring
+real(r8), parameter :: MIN_VALUE = 0.0
+real(r8), parameter :: MAX_VALUE = 0.1
+
+contains
+
+!----------------------------------------------------------------------
+
+subroutine initialize_module
+
+call register_module(source, revision, revdate)
+module_initialized = .true.
+
+end subroutine initialize_module
+
+!----------------------------------------------------------------------------
+
+subroutine get_expected_absolute_humidity(state_handle, ens_size, location, ah, istatus)
+
+type(ensemble_type), intent(in)     :: state_handle
+integer,             intent(in)     :: ens_size
+type(location_type), intent(in)     :: location
+real(r8),            intent(out)    :: ah(ens_size)    ! absolute humidity
+integer,             intent(out)    :: istatus(ens_size)
+
+real(r8), dimension(ens_size) :: qvap, tmpk, pres, rho_dry, tmpv
+integer,  dimension(ens_size) :: qvap_istatus, tmpk_istatus, pres_istatus
+real(r8)                      :: xyz(3)
+logical :: return_now
+
+if ( .not. module_initialized ) call initialize_module
+
+istatus = 0
+
+!  interpolate the mixing ratio to the location
+call interpolate(state_handle, ens_size, location, QTY_VAPOR_MIXING_RATIO, qvap, qvap_istatus)
+call track_status(ens_size, qvap_istatus, ah, istatus, return_now)
+
+where (istatus == 0 .and. qvap < 0.0_r8) qvap = epsilon(0.0_r8)  ! clamping?
+where (istatus /= 0) istatus = 99
+if(return_now) return
+
+!  interpolate the temperature to the desired location
+call interpolate(state_handle, ens_size, location, QTY_TEMPERATURE, tmpk, tmpk_istatus)
+call track_status(ens_size, tmpk_istatus, ah, istatus, return_now)
+
+where (tmpk <= 0.0_r8)
+   ah = missing_r8
+   istatus = 99
+endwhere
+
+if(all(istatus /= 0)) return  ! not using return_now because where clause modifies istatus.
```

Comment:
I suspect you copied this from a file that had this comment.  Since this is a new file, the comment is not correct and should be removed.  If you want to add a comment before this line, you could explain what is happenning next:
```
! If any of the interpolations failed, just return now
```